tests/explainers/test_tree.py

```diff
@@ -1171,6 +1171,10 @@ def test_xgboost_dmatrix_propagation(self, Clf):
         shap_values = explainer.shap_values(X_nan)
         # check that SHAP values sum to model output
         assert np.allclose(margin, explainer.expected_value + shap_values.sum(axis=1))
+        interaction_values = explainer.shap_interaction_values(X_nan)
```

Comment:
are your code changes tested by this? I set a breakpoint and this test doesn't run into `encode_array_if_needed`